sherpa/astro/instrument.py

```diff
@@ -1432,3 +1432,57 @@ def calc_grp_chan_matrix(fname):
     except sherpa.utils.err.IOErr as ioerr:
         print(ioerr)
         raise ioerr
+
+
+def has_pha_response(model):
```

Comment:
That's an interesting idea. I might use that heuristic in #1182, as well to get rid of the "is_source" parameter. Not sure, but I'm definitely thinking about it.